Dietpi-Software: add BirdNET-Go

mtekman opened the pull request:
Add BirdNET-Go, a realtime bird call monitoring webserver using embedded AI models

https://github.com/tphakala/birdnet-go

Inspired by this blog post: https://hannahilea.com/blog/birdnet-intro/

Notes:
* It's relatively low footprint on my RPi5 (0.7% passive CPU usage)
* Dependencies list sqlite3, but it could easily be mariadb, it just needs a config switch.
* The install consists solely of binaries, from FOSS software, but still it's pure binaries.
* Supported arches: amd64 and arm64
* It's listed under "Cameras" despite being an audio monitoring tool

Once installed you need to navigate to ~http://localhost:8127/ui (there's an [ongoing bug](https://github.com/tphakala/birdnet-go/issues/1045) with the root URL)~ http://localhost:8127

mtekman committed:
dietpi-software: add birdnet-go

mtekman committed:
shellcheck fix

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -9240,6 +9247,83 @@ _EOF_
+			## Move binaries to final destination
+			G_EXEC mkdir "${bnet_install}"
+			G_EXEC mv birdnet/birdnet-go "${bnet_install}"/birdnet-go
+			G_EXEC mv birdnet/libtensorflowlite_c.so "${bnet_install}"/libtensorflowlite_c.so
```
Available on Debian from Forky on: https://packages.debian.org/forky/libtensorflow-lite2.14.1

Not really relevant yet, but to keep in mind.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -9240,6 +9247,83 @@ _EOF_
+			# - $HOME/.config/birdnet-go/config.yaml (default)
+			# - /etc/birdnet-go/config.yaml
+			## initial call to generate config
+			LD_LIBRARY_PATH="${bnet_install}:$LD_LIBRARY_PATH" "${bnet_install}"/birdnet-go >/dev/null
```
Looks like something that should be error-handled:
```suggestion
			LD_LIBRARY_PATH="${bnet_install}:$LD_LIBRARY_PATH" G_EXEC "${bnet_install}"/birdnet-go
```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -9240,6 +9247,83 @@ _EOF_
+			## initial call to generate config
+			LD_LIBRARY_PATH="${bnet_install}:$LD_LIBRARY_PATH" "${bnet_install}"/birdnet-go >/dev/null
+
+			local tmp_config=${HOME}/.config/birdnet-go/config.yaml
```
Maybe there is a way to create the config at `/etc/birdnet-go/config.yaml` directly?

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -9240,6 +9247,83 @@ _EOF_
+			G_EXEC mkdir /etc/birdnet-go
+			G_EXEC mv "${tmp_config}" /etc/birdnet-go/config.yaml	   # move config to hardcoded /etc path
+
+			Create_User -g dietpi -d "${bnet_data}" birdnet
```
Does it need to have `dietpi` group permissions for a particular reason? Otherwise I'd just skip that and let it have its own group.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -9240,6 +9247,83 @@ _EOF_
 
+		if To_Install 127 birdnet-go # BirdNET-Go
+		then
+			local bnet_release="nightly-20250904"
```
I guess you use the nightly for a reason. The latest one could be pulled from the GitHub API as well: https://api.github.com/repos/tphakala/birdnet-go/releases

mtekman replied:
I just upgraded to a new nightly, but I think it should be pinned to something just in case something changes upstream.

I can't seem to compile the source, so I have a feeling that dev is a bit fast and loose.

MichaIng replied:
Did you try to compile it with `make`, or did you use `task` as advised at the top of the `Makefile`?

Here a related warning was added: https://github.com/tphakala/birdnet-go/commit/96ff58e

mtekman replied:
Yep, task. Kept hitting compile issues. Please let me know if you're seeing different

MichaIng replied:
Okay, anyway makes probably sense to stay a a fixed nightly until there is a new stable release. Then we can switch to latest stable releases, which should change sufficiently rarely that we can review new releases that can be seen in our URL update PRs #7780, and in case run test installs if there was a larger or breaking change.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -9240,6 +9247,83 @@ _EOF_
+			local bnet_release="nightly-20250904"
+			local bnet_http_port=8127
+			local bnet_install=/opt/birdnet-go
+			local bnet_log=/var/log
```
```suggestion
			local bnet_log=/var/log/birdnet
```
In the systemd unit, `LoggindDirectory=birdnet` can be used to have that directory created with correct owner automatically.

mtekman committed:
Update dietpi/dietpi-software

Co-authored-by: MichaIng <micha@dietpi.com>

mtekman committed:
Update dietpi/dietpi-software

Co-authored-by: MichaIng <micha@dietpi.com>

mtekman committed:
change /opt and /mnt from birdnet-go to birdnet

mtekman committed:
use config file in home dir only

mtekman committed:
remove /etc/birdnet-go mentions

mtekman committed:
create user before executing birdnet for firstime

mtekman committed:
upgrade release, use birdnet group

mtekman committed:
better title, and divert to newui by default to fix css problem

MichaIng committed:
dietpi-software: BirdNET-Go: several chore changes

* Added service and port check to install test workflow
* Added version update to update_url workflow
* Added support to dietpi-survey report page
* Added changelog entry
* Added dietpi-services support
* Added GitHub link to readme credits section
* Hide install option on unsupported architectures
* Added official install scripts link as code section comment, for reference if needed
* Apply consistent quotation, variable call syntax and indentation
* Align bnet_* variable name lengths, now all 9 characters long, yey xD
* Install yq permanently, it might be used by the admin already
* Install and use patchelf to add the current dir as "RUNPATH", making "LD_LIBRARY_PATH" obsolete
* Remove the old install dir on reinstall
* Touch config only on fresh installs
* Apply settings with a single yq call
* Add systemd service hardening, limit restarts, and assure logging dir is created by the unit
* Fix logging dir removal on uninstall

MichaIng committed:
dietpi-software: BirdNET-Go: disable on Bullseye

Also, install yq only if needed, and purge it afterwards.

MichaIng committed:
dietpi-software: BirdNET-Go: remove user on uninstall

and apply absolute database path. It works without since the wirking directory is set in the systemd unit, but we better be failsafe and set all known paths absolute.

MichaIng committed:
dietpi-software: BirdNET-Go: fix config variable

MichaIng committed:
dietpi-software: BirdNET-Go: service updates

Permit write access to the (private) /tmp dir. But remove write access from its install dir.

MichaIng committed:
dietpi-software: BirdNET-Go: remove  /var/log/birdnet

and move install code to end of function, so "G_AGP yq" does no autopurge Python packages if installed by any other software option in the same run.

MichaIng committed:
dietpi-software: BirdNET-Go: fix yq purge logic

MichaIng reviewed (approved):
Ready from my end. Last install tests also went through well, including service and network port check: https://github.com/MichaIng/DietPi/actions/runs/19105759477
